gcc default OpenMP

TorreZuk opened the pull request:
* use old system for gcc build

TorreZuk committed:
gcc default OpenMP

estewart08 commented on clients/CMakeLists.txt:
```diff
@@ -134,9 +136,11 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
+  endif()
 
-  if (TARGET OpenMP::omp)
+  if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND TARGET OpenMP::omp)
```
This "clang check" here may be a bit repetitive as the only way OpenMP::omp is set if line 140 is executed.

TorreZuk replied:
> This "clang check" here may be a bit repetitive as the only way OpenMP::omp is set if line 140 is executed.

Sure but like to be clear when I expect to going into the if

estewart08 commented on clients/CMakeLists.txt:
```diff
@@ -59,20 +59,22 @@ function( get_cblas cblas_libs cblas_inc )
-    set_target_properties( ${lib_target_} PROPERTIES
-      INSTALL_RPATH "$ORIGIN/../llvm/${openmp_LIB_DIR}"
-    )
+  if( CMAKE_CXX_COMPILER_ID MATCHES "Clang"  )
```
We cannot guard `TARGET OpenMP::OpenMP_CXX` with `Clang` because below we are using -L /path/to/rocm/lib -lomp which ldd shows libomp.so -> not found. We would need to move the Clang guard down to line 70.

`elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND TARGET OpenMP::omp)`

TorreZuk replied:
See if later revisions covers this now, the rpaths are all llvm/clang properties so think fine to do nothing for gcc libgomp.

TorreZuk committed:
OpenMP for clang (old and new config) and gcc

estewart08 committed:
Add Clang guards around usage of openmp config (#1050)

This fixes a gcc issue where clang headers are being brought in by
the openmp-config.

Additional cleanup:
Remove unnecessary options added to COMMON_LINK_LIBS. If the openmp
config is not found then OpenMP::OpenMP_CXX will be the fallback for
both gcc and clang.

(cherry picked from commit 46391031ec86a8edc7874d7dc8f35c4923134f05)
